hedera-state-validator/src/main/java/com/hedera/statevalidation/exporters/JsonExporter.java

```diff
@@ -107,28 +112,48 @@ private void processRange(String fileName, long start, long end) {
                 final StateKey stateKey;
                 final StateValue stateValue;
                 try {
-                    stateKey = StateKey.PROTOBUF.parse(keyBytes);
-                    if (expectedStateId != -1
-                            && expectedStateId != stateKey.key().kind().protoOrdinal()) {
+                    int tag = keyBytes.getVarInt(0, false);
+                    // normalize stateId for singletons
+                    final int actualStateId = tag >> TAG_FIELD_OFFSET  == 1 ?
+                            keyBytes.getVarInt(1, false) : tag >> TAG_FIELD_OFFSET;
```

Comment:
To avoid dealing with positions and tag sizes, readVarInt() should be used instead of getVarInt()